src/System/Taffybar.hs

```diff
@@ -258,6 +260,13 @@ useMonitorNumber c@(cfg, _) = return umn
             | mnumber == monitorNumber cfg = Just c
             | otherwise = Nothing
 
+-- | Use the primary monitor as set by Xrandr.
+usePrimaryMonitor :: TaffybarConfigEQ -> IO (Int -> Maybe TaffybarConfigEQ)
+usePrimaryMonitor c@(cfg, _) = do
+  maybePrimary <- withDefaultCtx getPrimaryOutputNumber
```

Comment:
Just a note: using withDefaultCtx too liberally, which actually opens a new X11 connection is what led to #188, and forced me to write #189.

In this case, I think its fine, since this function will only be called at startup, and any time taffybar recalculates which monitors it should display on, which is not going to be very often. You don't really have a way to access the pager x11 connection anyway, so not much can be done, I suppose.